lib/convert/src/impls/servers.rs

```diff
@@ -0,0 +1,115 @@
+use std::collections::HashMap;
+
+use proto::backend;
+use rivet_api::models;
+use rivet_operation::prelude::*;
+
+use crate::{ApiFrom, ApiTryFrom};
+
+impl ApiFrom<models::ServersResources> for backend::servers::ServerResources {
+	fn api_from(value: models::ServersResources) -> backend::servers::ServerResources {
+		backend::servers::ServerResources {
+			cpu_core_numerator: value.cpu_core_numerator,
+			cpu_core_denominator: value.cpu_core_denominator,
+			memory_megabytes: value.memory_megabytes,
+		}
+	}
+}
+
+impl ApiTryFrom<models::ServersRuntime> for backend::pkg::servers::server_create::request::Runtime {
+	type Error = GlobalError;
+
+	fn api_try_from(
+		value: models::ServersRuntime,
+	) -> GlobalResult<backend::pkg::servers::server_create::request::Runtime> {
+		match (value.docker,) {
+			(Some(docker_runtime),) => Ok(
+				backend::pkg::servers::server_create::request::Runtime::DockerRuntime(
+					backend::servers::DockerRuntime {
+						args: docker_runtime.args,
+						environment: docker_runtime.environment,
+						image_id: Some(docker_runtime.image_id.into()),
+						network: Some(unwrap!(backend::servers::DockerNetwork::api_try_from(
+							*docker_runtime.network,
+						))),
+					},
+				),
+			),
+			(None,) => bail_with!(SERVERS_NO_RUNTIME),
+			// This needs to be added if other runtimes are added
+			// _ => bail_with!(SERVERS_MULTIPLE_RUNTIMES),
+		}
+	}
+}
+
+impl ApiTryFrom<models::ServersDockerNetwork> for backend::servers::DockerNetwork {
+	type Error = GlobalError;
+
+	fn api_try_from(
+		value: models::ServersDockerNetwork,
+	) -> GlobalResult<backend::servers::DockerNetwork> {
+		Ok(backend::servers::DockerNetwork {
+			mode: backend::servers::DockerNetworkMode::api_from(value.mode) as i32,
+			ports: unwrap!(value
+				.ports
+				.into_iter()
+				.map(|(s, p)| Ok((s, unwrap!(backend::servers::DockerPort::api_try_from(p)))))
```

Comment:
Shoould be able to just do 
```suggestion
				.map(|(s, p)| Ok((s, p.api_try_into()?)))
```
with rust type inference

Reply:
Ah dope, will try to simplify where I can :+1: